CharlieBackend.Data/Repositories/Impl/AttachmentRepository.cs

```diff
@@ -14,6 +18,41 @@ public AttachmentRepository(ApplicationContext applicationContext)
         {
         }
 
+        public async Task<List<AttachmentDto>> GetAttachmentList(long accountId, long? courseId, long? groupId, long? requestedStudentAccountId, 
+                                                                DateTime? startDate, DateTime? finishDate)
+        {
+            return await _applicationContext.Attachments
+                    .WhereIf(_applicationContext.Accounts.Find(accountId).Role == UserRole.Mentor, x => _applicationContext.Mentors
```

Comment:
You might get null reference exception if account is not found(Find return entity or null)